src/OneExplorer/OneExplorer.ts

```diff
@@ -185,11 +185,26 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
       warningMessage = 'WARNING: Renaming may result in some unexpected changes on the tree view.';
     }
 
+    const validateInputPath = (newname: string): string|undefined => {
+      const oldpath = oneNode.node.path;
+      const dirpath = path.dirname(oneNode.node.uri.fsPath);
+      const newpath: string = `${dirpath}/${newname}`;
+
+      if (path.extname(newpath) !== path.extname(oldpath)) {
+        return `File ext must be (${path.extname(oldpath)})`;
+      }
+
+      if (fs.existsSync(newpath)) {
+        return `Invalid: Path already exists!`;
```

Comment:
```suggestion
        return `Invalid: File already exists!`;
```

It looks not to renaming directory. So, File already exists would be more clear for user IMO